modules/filters/mod_xml2enc.c

```diff
@@ -332,13 +332,17 @@ static apr_status_t xml2enc_ffunc(ap_filter_t* f, apr_bucket_brigade* bb)
         return ap_pass_brigade(f->next, bb) ;
     }
 
-    ctype = apr_pstrdup(f->r->pool, f->r->content_type);
-    for (p = ctype; *p; ++p)
-        if (isupper(*p))
-            *p = tolower(*p);
-
-    /* only act if starts-with "text/" or contains "xml" */
-    if (strncmp(ctype, "text/", 5) && !strstr(ctype, "xml"))  {
+    /* Extract the media type, ignoring parameters in content-type. */
+    mtype = apr_pstrdup(f->r->pool, f->r->content_type);
+    if ((p = ap_strchr(mtype, ';')) != NULL) *p = '\0';
+    ap_str_tolower(mtype);
+
+    /* Accept text/ types, plus any XML media type per RFC 7303. */
+    if (!(strncmp(mtype, "text/", 5) == 0
```

Comment:
@notroj Is this correct? It looks like it mixes two flavors of negating strncmp result.

Reply:
I struggled to make this readable, hoped it was easier after De Morgan-ing it, but it's still somehow not...

intuition: if you look just at the first clause it turns `if strncmp() != 0` into `if !(strncmp() == 0)` which is equivalent

Reply:
I see, I just realized the block inside is removing the filter, not taking action on the body.